OpenSim/Common/Component.cpp

```diff
@@ -457,8 +479,8 @@ int Component::getNumStateVariables() const
     //Get the number of state variables added (or exposed) by this Component
     int ns = getNumStateVariablesAddedByComponent(); 
     // And then include the states of its subcomponents
-    for(unsigned int i=0; i<_components.size(); i++)
-        ns += _components[i]->getNumStateVariables();
+    for(unsigned int i=0; i<_propertySubcomponents.size(); i++)
+        ns += _propertySubcomponents[i]->getNumStateVariables();
```

Comment:
Good find! It most definitely should have to go through the other two! 
I was waiting to replace with a "my subcomponents" iterator instead of looping through three lists. I may be able use that in multiple places see above. Maybe I will just make it correct first then replace with the iterator.